atkinson/pungi/depends.py

```diff
@@ -0,0 +1,308 @@
+        ret.append(info)
+    return ret
+
+
+def dep_to_tuple(dep):
+    """Convert one of our dependencies into an RPM-style tuple"""
+    if not isinstance(dep, dict):
+        raise ValueError('dep is not a dict')
+    if 'version' not in dep:
+        raise ValueError('Cannot convert {0}: version missing'.format(str(dep)))
+    epoch = 0
+    version = None
+    release = None
+    if 'epoch' in dep:
+        epoch = int(dep['epoch'])
+    version = dep['version']
+    if 'release' in dep:
+        release = dep['release']
+    return (epoch, version, release)
+
+
+def dep_to_nevr(dep):
+    """Convert one of our dependencies into an RPM NEVR string"""
+    if not isinstance(dep, dict):
+        raise ValueError('dep is not a dict')
+    for field in ('name', 'version', 'release'):
+        if field not in dep:
+            raise ValueError('Cannot convert {0}: {1} missing'.format(str(dep), field))
+    ret = dep['name']
+    ret = ret + '-'
+    if 'epoch' in dep and dep['epoch'] not in ('0', 0, ''):
+        ret = ret + str(dep['epoch']) + ':'
+    ret = ret + str(dep['version'])
+    ret = ret + '-' + dep['release']
+    return ret
+
+
+def dep_to_rpmdep(dep):
+    """Convert a dependency into an RPM dependency string"""
+    if 'comparison' not in dep:
+        return dep['name']
+    ret = '{0} {1} '.format(dep['name'], dep['comparison'])
+    if 'epoch' in dep and dep['epoch'] not in ('', 0, '0'):
+        ret = ret + '{0}:'.format(dep['epoch'])
+    ret = ret + '{0}'.format(dep['version'])
+    if 'release' in dep and dep['release'] != '':
+        ret = ret + '-{0}'.format(dep['release'])
+    return ret
+
+
+def rpmdep_to_dep(dep):
+    """Convert a RPM style dependency into a simple dict"""
+    vals = dep.split(' ')
+    if len(vals) == 1:
+        return {'name': vals[0]}
+    if len(vals) != 3:
+        raise ValueError('Could not parse: \"' + dep + '\"')
+    ret = {'name': vals[0],
+           'comparison': vals[1]}
+    evr = vals[2].split('-')
+    if len(evr) > 2:
+        raise ValueError('Could not parse: \"' + dep + '\"')
+
+    if len(evr) == 2:
+        ret['release'] = evr[1]
+        if ret['release'] == 'None':
+            del ret['release']
+
+    if ':' in evr[0]:
+        e_v = evr[0].split(':')
+        ret['epoch'] = int(e_v[0])
+        if ret['epoch'] == '0':
+            del ret['epoch']
+        ret['version'] = e_v[1]
+    else:
+        ret['version'] = evr[0]
+
+    return ret
+
+
+def nevr_to_dep(dep):
+    """Convert a RPM NEVR string to a simple dict"""
+    (name, version, release, epoch, _) = split_filename(dep)
+    ret = {}
+
+    if name == '':
+        raise ValueError('Could not parse: \"' + dep + '\"')
+
+    ret['name'] = name
+    ret['version'] = version
+    ret['comparison'] = '=='
+    if epoch not in ('', 0, '0'):
+        ret['epoch'] = int(epoch)
+    if release != 'None' and release is not None:
+        ret['release'] = release
+
+    return ret
+
+
+def _main(argv):
```

Comment:
At some point we should remove this. But that can be at a later time when we have more of the runner infrastructure setup. We could however pull this out to a contrib or manual test area that just calls this module.